File: Progammers_가장 큰 수.cpp
#include <vector>
#include <string>
#include <algorithm>
using namespace std;
 
/* return 1 if a + b is bigger*/
bool cmp(string a, string b) {
    return a + b > b + a;
}
 
string solution(vector<int> numbers) { 
	string answer = "";
	vector<string> tmp;
    
	for (int i = 0; i < numbers.size(); i++) 
    {
        tmp.push_back(to_string(numbers[i]));
    }
    
	sort(tmp.begin(), tmp.end(), cmp); // sort in func cmp 
    
	for (int i = 0; i < tmp.size(); i++)
    {
        answer += tmp[i]; 
    }
    
    /* handling expection like 00 */
	if (answer[0] == '0') return "0"; 
    
	return answer;
}

/* revised from https://mungto.tistory.com/22 */

File: Progammers_가장 큰 수.py
import functools

def comparator(a, b):
    t1 = a + b
    t2 = b + a
    
    # if t1 is bigger 1, else if t2 is bigger 2, else if they're same 0
    return (int(t1) > int(t2)) - (int(t1) < int(t2)) 

def solution(numbers):
    n = [str(x) for x in numbers]   # make every num into string
    n = sorted(n, key = functools.cmp_to_key(comparator), reverse=True)
    answer = str(int(''.join(n)))
    
    return answer

# revised from https://eda-ai-lab.tistory.com/467
